src/client/components/Headline/Headline.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import styled, { StyledComponent } from '@emotion/styled'
+import { css } from '@emotion/core'
+import { fonts, colorsV3 } from '@hedviginsurance/brand'
+import { MEDIUM_SCREEN_MEDIA_QUERY } from 'utils/mediaQueries'
+
+const { FAVORIT } = fonts
+const { gray900, gray100 } = colorsV3
+
+export type Props = {
+  variant: 'xl' | 'l' | 'm' | 's' | 'xs' | 'overline'
+  headingLevel: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+  colorVariant: 'dark' | 'light'
+  children: React.ReactNode
+}
+
+type ColorProp = Pick<Props, 'colorVariant'>
+
+type StyleProps = {
+  as: Props['headingLevel']
+} & ColorProp
+
+type GetBaseStylesParams = ColorProp
```

Comment:
🤔 Maybe this type can be removed?! Maybe it can be simpler without it:

What do you think of this?
```
type ColorVariant = 'dark' | 'light'

export type Props = {
  variant: 'xl' | 'l' | 'm' | 's' | 'xs' | 'overline'
  headingLevel: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  colorVariant: ColorVariant
  children: React.ReactNode
}

type StyleProps = {
  as: Props['headingLevel']
  colorVariant: ColorVariant
}

const getBaseStyles = (colorVariant: ColorVariant) => {
  const color = colorVariant === 'light' ? gray100 : gray900

  return css`
    margin: 0;
    padding: 0;
    font-family: ${FAVORIT};
    font-weight: 400;
    color: ${color};
  `
}
```

Reply:
OK so I got rid of the `GetBaseStylesParams` type since it was just the same as `ColorProp` type. But I decided to not go with this suggestion since I generally like having "named" parameter for functions (i.e. passing an object as a single parameter). This is of course more beneficial when having multiple function parameters (which we don't in this case) but it's an easier principle to follow to just always do it like that than changing it when we want to add more parameters. And no, I'm not super consistent with this haha 🙈 But I try!